client/src/http_default.rs

```diff
@@ -39,50 +43,52 @@ impl HttpReqExt for http::request::Builder {
     }
 }
 
-pub fn post<U, P, K, V>(
-    url: U,
-    body: Bytes,
-    query_params: P,
-    headers: Headers,
-) -> Result<Response<Bytes>>
+/// Default request builder & sender implemented on top of `attohttpc` crate.
+pub struct DefaultRequestBuilder<T: Body = attohttpc::body::Bytes<Bytes>>(
+    AttoHttpRequestBuilder<T>,
+);
+
+impl<T> DefaultRequestBuilder<T>
 where
-    U: AsRef<str>,
-    P: IntoIterator,
-    P::Item: Borrow<(K, V)>,
-    K: AsRef<str>,
-    V: ToString,
+    T: Body,
 {
-    let url = url.as_ref();
-    let response = attohttpc::post(url)
-        .bytes(body)
-        .params(query_params)
-        .set_headers(headers)?
-        .send()
-        .wrap_err_with(|| format!("Failed to send http post request to {}", url))?;
-    ClientResponse(response).try_into()
+    pub fn send(self) -> Result<Response<Bytes>> {
```

Comment:
Need documentaion.
Actually why don't our lints catch that?

Reply:
I guess the reason is that it is a part of non-public module. You can also notice some other public stuff in this module that wasn't documented previously too.